### PR TITLE
Added unit tests for athena-sqlserver.

### PR DESCRIPTION
**Pull request by ritiktrianz**
*Issue #, if available:*

*Description of changes:*
Added unit tests for athena-sqlserver connector.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/resolver/SQLServerJDBCCaseResolverTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            ResultSet tableResultSet = mockResultSet(tableCols, tableTypes, tableData, new AtomicInteger(-1));
+            when(preparedStatement.executeQuery()).thenReturn(tableResultSet);
+
+            String adjustedTableName = resolver.getAdjustedTableNameString(mockConnection, schemaName, tableName, Map.of(
+                    CASING_MODE_CONFIGURATION_KEY, CaseResolver.FederationSDKCasingMode.CASE_INSENSITIVE_SEARCH.name()));
+            assertEquals(tableName.toUpperCase(), adjustedTableName);
+    }
+
+    @Test
+    public void getAdjustedTableNameObject_caseInsensitiveSearchOnObject_returnsAdjustedTableNameObject() throws SQLException {
```
Comment (fal-bharadwaj) [inline athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/resolver/SQLServerJDBCCaseResolverTest.java:89]: Should we add more tests for the following - 
- case-sensitive use-cases
- edge cases for CASING_MODE_CONFIGURATION_KEY (null/invalid values)

File: athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/SqlServerEnvironmentPropertiesTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    public void connectionPropertiesToEnvironment_WithValidProperties_ReturnsExpectedConnectionString()
+    {
+        Map<String, String> sqlServerConnectionProperties = sqlServerEnvironmentProperties.connectionPropertiesToEnvironment(connectionProperties);
+
+        String expectedConnectionString = "sqlserver://jdbc:sqlserver://test.sqlserver.com:1433;databaseName=testdb;${sqlserver-secret}";
+        assertEquals(expectedConnectionString, sqlServerConnectionProperties.get(DEFAULT));
+    }
+    
+    @Test
+    public void getDelimiter_ReturnsSemicolonDelimiter()
```
Comment (fal-bharadwaj) [inline athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/SqlServerEnvironmentPropertiesTest.java]: let's rename to "getDelimiter_WithDefaultConfiguration_ReturnsSemicolonDelimiter()"

File: athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/SqlServerEnvironmentPropertiesTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import static com.amazonaws.athena.connector.lambda.connection.EnvironmentConstants.SECRET_NAME;
+import static org.junit.Assert.assertEquals;
+
+public class SqlServerEnvironmentPropertiesTest
+{
+    private Map<String, String> connectionProperties;
+    private SqlServerEnvironmentProperties sqlServerEnvironmentProperties;
+
+    @Before
+    public void setUp()
```
Comment (fal-bharadwaj) [inline athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/SqlServerEnvironmentPropertiesTest.java:41]: let's test with empty connection properties and missing required params like HOST/PORT etc.

File: athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/SqlServerMetadataHandlerTest.java [hunk 4]
```diff
@@ -453,7 +464,217 @@ public void doListSchemaNames()
...
+                    {Types.TIMESTAMP, 0, "col_datetime2", 0, 0}, // simulate DATETIME2
+                    {Types.TIMESTAMP, 0, "col_smalldatetime", 0, 0}, // simulate SMALLDATETIME
+                    {Types.TIMESTAMP_WITH_TIMEZONE, 0, "col_datetimeoffset", 0, 0}
+            };
+
+            AtomicInteger rowNumber = new AtomicInteger(-1);
+            ResultSet resultSet = mockResultSet(schema, values, rowNumber);
+
+            // Mocking sys.columns response (DATA_TYPE column as string names from enum)
+            ResultSet columnTypesResultSet = mockResultSet(
```
Comment (fal-bharadwaj) [inline athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/SqlServerMetadataHandlerTest.java:491]: can some of this mock setup (result set, connection object below etc) be generalized into a helper method cuz its used in multiple places.

File: athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/resolver/SQLServerJDBCCaseResolverTest.java (hunk 1 above)
Comment (ritiktrianz) [reply athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/resolver/SQLServerJDBCCaseResolverTest.java:89]: Hi @fal-bharadwaj, thanks for the review.

As suggested, added more tests to cover case-sensitive use-cases, edge cases for CASING_MODE_CONFIGURATION_KEY, etc.

File: athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/SqlServerEnvironmentPropertiesTest.java (hunk 2 above)
Comment (ritiktrianz) [reply athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/SqlServerEnvironmentPropertiesTest.java]: Hi @fal-bharadwaj , renamed test name to "getDelimiter_WithDefaultConfiguration_ReturnsSemicolonDelimiter()".

File: athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/SqlServerEnvironmentPropertiesTest.java (hunk 3 above)
Comment (ritiktrianz) [reply athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/SqlServerEnvironmentPropertiesTest.java:41]: Hi @fal-bharadwaj  , added more tests to cover these scenarios.

File: athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/SqlServerMetadataHandlerTest.java (hunk 4 above)
Comment (ritiktrianz) [reply athena-sqlserver/src/test/java/com/amazonaws/athena/connectors/sqlserver/SqlServerMetadataHandlerTest.java:491]: Hi @fal-bharadwaj  , added helper method for common code.
